features/gm.ts

```diff
@@ -81,23 +87,62 @@ export class GroupMessageController implements GroupMessageGetter, GroupMessageL
         return event;
     }
 
-    async prepareGroupMessageEvent(groupAddr: PublicKey, text: string) {
+    async prepareGroupMessageEvent(groupAddr: PublicKey, message: string | Blob) {
+        const eventsToSend: NostrEvent<NostrKind.Group_Message, Tag>[] = [];
         const groupCtx = this.getGroupChatCtx(groupAddr);
         if (groupCtx == undefined) {
             return new Error(`group ctx for ${groupAddr.bech32()} is empty`);
         }
-        const nostrEvent = await prepareEncryptedNostrEvent(this.ctx, {
-            content: JSON.stringify({
-                type: "gm_message",
-                text,
-            }),
-            kind: NostrKind.Group_Message,
-            tags: [
-                ["p", groupAddr.hex],
-            ],
-            encryptKey: groupCtx.publicKey,
-        });
-        return nostrEvent;
+        let text: string;
+        let kind: "text" | "image";
+        if (message instanceof Blob) {
+            kind = "image";
+            text = await blobToBase64(message);
+        } else {
+            kind = "text";
+            text = message;
+        }
+        const event = await prepareEncryptedNostrEvent(
+            this.ctx,
+            {
+                content: JSON.stringify({
+                    type: "gm_message",
+                    text: text,
```

Comment:
Now this is not text anymore, semantically speaking. But before we have binary format for Nostr event, we can't do much about it.